liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,256 @@
+        (0x1b1c, 0x0c15, None, 'Corsair Hydro H100i Pro', {}),
+        (0x1b1c, 0x0c16, None, 'Corsair Hydro H80i Pro', {}),
+    ]
+
+    def _write(self, data):
+        """Write data to the AIO and log"""
+        _LOGGER.debug('write %s', LazyHexRepr(data))
+        self.device.write(_PRO_WRITE_ENDPOINT, data, _WRITE_TIMEOUT)
+
+    # Not calling this (or begin transaction) since they do not seem to be needed for H100i Pro
+    def _end_transaction_and_read(self, length=_READ_LENGTH):
+        """End the transaction by reading from the device.
+        According to the official documentation, as well as Craig's open-source
+        implementation (libSiUSBXp), it should be necessary to check the queue
+        size and read data in chunks.  However, leviathan and its derivatives
+        seem to work fine without this complexity; we currently try the same
+        approach.
+        """
+        msg = self.device.read(_PRO_READ_ENDPOINT, length, _READ_TIMEOUT)
+        _LOGGER.debug('received %s', LazyHexRepr(msg))
+        self.device.release()
+        return msg
+
+    def _write_color_change_speed(self, mode, speed):
+        """Send the speed to cycle colors on the RGB pump"""
+        if speed < 1 or speed > 3:
+            raise ValueError('Unsupported speed ({}), only 1-3 supported'.format(speed))
+
+        self._write([0x53, _PRO_SPEEDS[mode][int(speed)-1]])
+        self._end_transaction_and_read(3)
+
+    def _write_color(self, colors):
+        """Write a list of colors to cycle between, the cycle pattern comes from the
+        value used for cycle speed"""
+        colorCount = 0
+        setColors = []
+        for color in colors:
+            colorCount = colorCount + 1
+            setColors = setColors + color
+        self._write([_PRO_CMD_WRITE_COLOR_LIST, colorCount] + setColors)
+        self._end_transaction_and_read(3)
+
+    def _get_fan_speeds(self):
+        """Read the RPM speed of the fans. Try to get the speeds for 3 fans as that is
+        how many the H150i PRO supports"""
+        speeds = []
+        for i in [0, 1, 2]:
+            self._write([_PRO_CMD_READ_FAN_SPEED, i])
+            msg = self._end_transaction_and_read(6)
+            if msg[0] != 0x41 or msg[1] != 0x12 or msg[2] != 0x34 or msg[3] != i:
+                _LOGGER.debug('Unable to get speed for fan id %d', i)
+                continue
+            speeds.append((i, (msg[4] << 8) + msg[5]))
+        return speeds
+
+    def _get_fan_indexes(self, channel):
+        if len(channel) > 3:
+            return [int(channel[3:]) - 1]
+        return [reading[0] for reading in self._get_fan_speeds()]
+
+    def initialize(self, pump_mode=None, **kwargs):
+        """Initialize the device."""
+        _LOGGER.debug('Pro configure device...')
+        if pump_mode == None:
+            return
+        else:
+            _LOGGER.debug('pump_mode %s', pump_mode)
+            self._write([_PRO_CMD_WRITE_PUMP_MODE, _PRO_PUMP_MODES.index(pump_mode)])
```

Comment:
The implicit comparison for `pump_mode` must be case insensitive.

I would also prefer to see all lowercase values in `_PRO_PUMP_MODES`.

Reply:
Done